### PR TITLE
list-ops: reimplement ambiguous tests

### PR DESCRIPTION
**Pull request by SleeplessByte**
The notation `(x, y) => x + y` for `foldl` and `foldr` is ambigious. Yes, there is a mathy standard for fold, but the fact is that this ordering is not consistent across languages, nor is it well known.

This attempts to make it completely un-ambigious what each parameter means, but since they _were_ ambiguous, it's technically a breaking change.

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]
Great!

**Review by SleeplessByte — COMMENTED** [review]
DON'T MERGE THIS YET. Need to delete `bin/configlet.exe`  which snuck in because of #1737 not yet being merged.

**Review by valentin-p — APPROVED** [review]

File: exercises/list-ops/canonical-data.json
Comment (petertseng) [inline exercises/list-ops/canonical-data.json:382]: I see that this is correct, though I'd like to note for my records that it needs the `/` to preserve fractions. Integer division will not work here, since it would compute `1 / 24 = 0`. I will take it into account when using this case.

File: exercises/list-ops/canonical-data.json
Comment (petertseng) [inline exercises/list-ops/canonical-data.json:489]: By my understanding of foldr, this should be 9 instead. Can you confirm?

* acc = 24, el = 4. new acc = 1/6
* acc = 1/6, el = 3. new acc = 18
* acc = 18, el = 2. new acc = 1/9
* acc = 1/9, el = 1. new acc = 9

File: exercises/list-ops/canonical-data.json
Comment (SleeplessByte) [reply exercises/list-ops/canonical-data.json:382]: Yes! I should add that as a comment.

I think it was the same with the previous case?

File: exercises/list-ops/canonical-data.json
Comment (SleeplessByte) [reply exercises/list-ops/canonical-data.json:489]: It's 9 in my test suite 🙈 . You're correct.

**Comment by SleeplessByte** [comment]
For the person merging: Do **not** squash.

File: exercises/list-ops/canonical-data.json
Comment (petertseng) [reply exercises/list-ops/canonical-data.json:382]: I think the previous case asked for integer division, since its expected value is zero and the only way to get a zero out of division is if you started with a zero. Since there is no zero present in the list nor in the initial value, I reason that the zero must have been introduced via integer division.

Comment (SleeplessByte) [reply exercises/list-ops/canonical-data.json:382]: Right. I couldn't get the test case to work in some languages I work in _because_ they do fractional division. I think commenting here is the right approach, and something we should have had on the previous case. I _will_ add it there too!

File: exercises/list-ops/canonical-data.json
Comment (petertseng) [inline exercises/list-ops/canonical-data.json:292]: note for posterity: The possible ambiguous interpretation for this function were:

* acc, el -> acc / el
    * acc: 5, el: 5. new acc 1
    * acc: 1, el: 2. new acc 0
* acc, el -> el / acc
    * acc: 5, el: 5. new acc 1
    * acc: 1, el: 2. new acc 2
 
So clearly the **latter** was intended. Note that this is different from the other case, where the **former** was intended. The fact that these two same strings require two different interpretations is what's confusing and is the motivation for this PR.

File: exercises/list-ops/canonical-data.json
Comment (petertseng) [inline exercises/list-ops/canonical-data.json:197]: note for posterity: The possible ambiguous interpretation for this function were:

* acc, el -> acc / el
    * acc: 5, el: 2. new acc 0
    * acc: 0, el: 5. new acc 0
* acc, el -> el / acc
    * acc: 5, el: 2. new acc 0
    * acc: 0, el: 5. division by zero.

So clearly the **former** was intended. Note that this is different from the other case, where the **latter** was intended. The fact that these two same strings require two different interpretations is what's confusing and is the motivation for this PR.

File: exercises/list-ops/canonical-data.json
Comment (petertseng) [inline exercises/list-ops/canonical-data.json:343]: The follow comment concerns **notation** only. It does not change the expected values of any test, or change the semantics of how `foldr` should operate.

I understand that some track maintainers may disagree about whether the function argument list should be **represented as a string** as `(acc, el)` or `(el, acc)`, but I would like to say to those track maintainers that I think we are just going to have to pick one and run with it. Again, I emphasise that either way there would not be a change to the operation of `foldr`: The implementing track should be sure to pass the element as `el` and the accumulator as `acc`, no matter the string representation here.

I advise implementing tracks to be aware of the conventions common to their language do what makes sense for their language. Why do I bring this up? Observe:

* Ramda (a functional library for Javascript): [foldr](https://ramdajs.com/0.9/docs/#foldr) is `(a, b -> a) -> a -> [b] -> a`. The passed function takes its accumulator first and the list element second, and `foldr` takes initial accumulator first and the list second.
* OCaml [List.fold_right](https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html#VALfold_right) is `('a -> 'b -> 'b) -> 'a list -> 'b -> 'b`. The passed function takes its list element first and the accumulator second, and `fold_right` takes the list first and accumulator second.
* Haskell [Prelude foldr](https://hackage.haskell.org/package/base-4.14.1.0/docs/Prelude.html#v:foldr) is `class Foldable t where foldr :: (a -> b -> b) -> b -> t a -> b`. The passed function takes its list element first and accumulator second, and `foldr` takes the initial accumulator first and the list second.

As you can see, I've listed three different languages doing things in three different ways, so I advise track maintainers looking to squabble over this that there is no universal convention, so we should just pick one for problem-specifications and run with it.

**Review by petertseng — APPROVED** [review]
The new cases add are all correct. I have added some notes for posterity.

**Comment by petertseng** [comment]
EBWODP:

> * acc, el -> acc / el
>    * acc: 5, el: 2. new acc 0

ah, my apologies. I meant to say new acc 2.

**Comment by petertseng** [comment]
By the way, I think the recommendation should change to be that these changes **should** be squashed. It will mean there will be no executable that got added and a subsequent commit removing it. And there are no formatting changes that need a separate commit, etc.

File: exercises/list-ops/canonical-data.json
Comment (SleeplessByte) [reply exercises/list-ops/canonical-data.json:343]: This is great advice and I agree 100% with it. I'll link this comment in the comments in the canonical data.

**Review by petertseng — APPROVED** [review]
just approving to let you know I saw the new advice added

**Pull request merged**
